state/address.go

```diff
@@ -192,7 +192,9 @@ func (st *State) filterHostPortsForManagementSpace(
 	}
 
 	var hostPortsForAgents []network.SpaceHostPorts
-	if mgmtSpace := config.JujuManagementSpace(); mgmtSpace != network.DefaultSpaceName {
+	// TODO (hml) 2019-10-29
+	// Investigate why we still need to check mgmtSpace != ""
+	if mgmtSpace := config.JujuManagementSpace(); mgmtSpace != "" {
```

Comment:
Removing this check causes the featuretests and state unit tests to blow up.  It will take longer than expected to resolve and might not make this pr.